src/modules/course/schema/enrolledCourse.schema.ts

```diff
@@ -0,0 +1,50 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+import * as mongoose from 'mongoose';
+
+export type EnrolledCourseDocument = EnrolledCourse & Document;
+
+interface video {
+  num: number;
+  timestamp?: Date;
+}
+
+@Schema()
+export class EnrolledCourse {
+  @Prop({ default: [].fill(false, 10) })
+  videos_watched: boolean[];
+
```

Comment:
How Will you refer to student?

Add a prop for studentId here.
and make the virtual somewhat like:
```
EnrolledCourseSchema.virtual('students', {
  ref: 'User',
  localField: 'studentId',
  foreignField: '_id',
});
```

Reply:
Ok